### PR TITLE
docs: explain express in crafting loopback 4

### LINKED ISSUE #1538: [Docs] Update `Why Not Express` section in Crafting LoopBack 4
## Description / Steps to reproduce / Feature proposal

From https://github.com/strongloop/loopback-next/issues/1362

The section https://loopback.io/doc/en/lb4/Crafting-LoopBack-4.html#why-not-express needs to be updated to reflect the fact that we have switched back to using Express -- explaining how it is used today (not for routing but only middleware ... while we still provide a sequence of actions) ... and the decision for this (access to existing middleware).

## Acceptance Criteria
- [ ] Update https://loopback.io/doc/en/lb4/Crafting-LoopBack-4.html#why-not-express

### PR DESCRIPTION
**Pull request by virkt25**
fixes #1538 

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: docs/site/Crafting-LoopBack-4.md [hunk 1]
```diff
@@ -421,44 +421,50 @@ independent of the nature of such artifacts.
...
-  instances either.
-
-- **Lack of declarative support**.
-
-  In Express, everything is done by JavaScript code as it works exactly as the
-  web site claims: `Fast, unopinionated, minimalist web framework for Node.js`.
-  In contrast, LoopBack is designed to facilitate API creation and composition
-  by conventions and patterns as best practices. More types of constructs are
-  introduced.
+### Why Express Still?
```
Comment (raymondfeng) [inline docs/site/Crafting-LoopBack-4.md]: Is it better to have `Why Express behind the scene?`

File: docs/site/Crafting-LoopBack-4.md [hunk 2]
```diff
@@ -421,44 +421,50 @@ independent of the nature of such artifacts.
...
+  > `app.use`.
+- Lack of declarative support
+  > In Express, everything is done by JavaScript ... In contrast, LoopBack is
+  > designed to facilitate API creation and composition by conventions and
+  > patterns as best practices.
+
+#### Circling back to Express with a twist
+
+LoopBack does a lot of things but our value add is the ease of use and data
+processing, not middleware and routing so we didn't want to reinvent the wheel
```
Comment (raymondfeng) [inline docs/site/Crafting-LoopBack-4.md]: Routing is the part we want to optimize based on controller metadata. As a result, we minimize the usage of Express routing.

File: docs/site/Crafting-LoopBack-4.md [hunk 3]
```diff
@@ -421,44 +421,50 @@ independent of the nature of such artifacts.
...
+  > register a middleware. The order of middleware is determined by the order of
+  > `app.use`.
+- Lack of declarative support
+  > In Express, everything is done by JavaScript ... In contrast, LoopBack is
+  > designed to facilitate API creation and composition by conventions and
+  > patterns as best practices.
+
+#### Circling back to Express with a twist
+
+LoopBack does a lot of things but our value add is the ease of use and data
```
Comment (shimks) [inline docs/site/Crafting-LoopBack-4.md]: Not sure what you mean by `but our value add is ...`

File: docs/site/Crafting-LoopBack-4.md [hunk 4]
```diff
@@ -421,44 +421,50 @@ independent of the nature of such artifacts.
...
+processing, not middleware and routing so we didn't want to reinvent the wheel
+by writing new middleware for LoopBack 4.
+
+The team explored leveraging
+[Express or Koa](https://github.com/strongloop/loopback-next/pull/1082) (but
+only for their middleware support). The final decision was to use Express in a
+way that bridges the gap by addressing the gaps identified above as follows:
+
+- LoopBack provides it's own
```
Comment (shimks) [inline docs/site/Crafting-LoopBack-4.md]: it's -> its

**Pull request merged**